pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -580,15 +611,12 @@ func (n *nodeLocalDNS) getHealthAddress() (healthAddress string) {
 }
 
 func (n *nodeLocalDNS) getAddress(useIPv6Brackets bool) string {
-	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
-	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+	if n.values.IPFamilies[0] == gardencorev1beta1.IPFamilyIPv4 {
 		return nodelocaldnsconstants.IPVSAddress
-	}
-	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+	} else {
```

Comment:
Can we please get rid of the `else`? There is no need for it anymore as we always fall through with IPv6.